### ISSUE TITLE
Model seems to return target values from input dictionary

### ISSUE THREAD

**Issue by daoiradrio** [atomistic-machine-learning/schnetpack#680]
Hi,

I am following the tutorial "Training a neural network on QM9" of the documentation (https://schnetpack.readthedocs.io/en/latest/tutorials/tutorial_02_qm9.html). When testing the trained model as shown in the first two cells of the section "Inference", it returns precisely the target values for U0 giving a total test loss of exactly 0. When instead I use the test routine of the PyTorch Lightning trainer, I get a non-zero test loss. Also, when I test the model as shown in the tutorial but with the difference that I remove the target values from the input dictionary before the forward pass, I get more realistic non-zero loss values. I'll attach the corresponding notebook, there I reduced the number of training/validation/test samples, feature channels and radial basis functions for a quick test run.

I would appreciate any kind of help pointing out what the problem might be or what I might be doing wrong.

Best,
Dario

[reproduction.zip](https://github.com/user-attachments/files/18053453/reproduction.zip)

**Comment by jnsLs** [comment]
Hi Dario, 

the reason for the confusion is that the target labels in the batch are overridden in inference.
That's why `y_pred[QM9.U0]` is the same as `batch[QM9.U0]`.

We are planning to change this but unfortunately it requires many changes in the code.

Best, Jonas
